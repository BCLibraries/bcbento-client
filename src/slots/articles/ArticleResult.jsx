import React from 'react';
import {PrimoRecordLink} from "../PrimoRecordLink";

function ArticleResult({article}) {
    return <li className="article-result-item media">
        <div className="media-body">
            <h3 className="article-result-item__media-heading media-heading">
                <a href={PrimoRecordLink(article)} className="article-result-item__title">
                    {article.title}
                </a>
            </h3>

            {article.date}

            <div className="article-result-item__creator">{article.creator}</div>
            <div className="article-result-item__publisher">
                {formatPartOf(article.isPartOf, article.journalTitle)}
            </div>

        </div>
    </li>;
}

// @TODO Replace
function formatPartOf(part_of, journal_title) {
    part_of = part_of.length > 0 ? part_of[0] : '';

    const parts = part_of.split(journal_title);

    if (!journal_title || parts.length !== 2) {
        return <span>{part_of}</span>
    }

    return <span className="article-result-item__journal-title">
            {parts[0]}<cite>{journal_title}</cite>{parts[1]}
        </span>
}

export default ArticleResult;