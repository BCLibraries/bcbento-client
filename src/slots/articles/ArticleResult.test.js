import renderer from 'react-test-renderer';
import ArticleResult from "./ArticleResult";

it('renders correctly', () => {
    const article = {
        id: "TN_cdi_proquest_journals_1656063155",
        title: "Otters",
        date: null,
        type: "article",
        dOI: null,
        creator: "Hart, Mark D",
        contributors: [],
        coverImages: [
            {
                url: "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?issn=0026-3451/MC.JPG&client=primo",
            },
            {
                url: "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?issn=0026-3451/MC.JPG&client=primo",
            }
        ],
        isPartOf: [
            "The Midwest quarterly (Pittsburg), 2014-06, Vol.55 (4), p.40"
        ],
        isPeerReviewed: true,
        isOpenAccess: false,
        journalTitle: [
            "The Midwest quarterly (Pittsburg)"
        ],
        libkeyAvailability: null,
    };

    const component = renderer.create(
        <ArticleResult article={article}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
