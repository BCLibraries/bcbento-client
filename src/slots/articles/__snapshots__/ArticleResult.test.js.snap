// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="article-result-item media"
>
  <div
    className="media-body"
  >
    <div
      className="article-result-item__type"
    >
      article
    </div>
    <h3
      className="article-result-item__media-heading media-heading"
    >
      <a
        className="article-result-item__title"
        href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=TN_cdi_proquest_journals_1656063155&context=PC&tab=pci_only&search_scope=pci&vid=bclib_new&lang=en_US"
        rel="noreferrer noopener"
        target="_blank"
      >
        Otters
      </a>
    </h3>
    <div
      className="article-result-item__creator"
    >
      Hart, Mark D
    </div>
    <div
      className="article-result-item__publisher"
    >
      <span
        className="article-result-item__journal-title"
      >
        <cite>
          The Midwest quarterly (Pittsburg)
        </cite>
        , 2014-06, Vol.55 (4), p.40
      </span>
    </div>
    <div
      className="article-result-item__peer_reviewed"
    >
      Peer Reviewed
    </div>
  </div>
</li>
`;
